### PR TITLE
Zero mean should produce a 1D output

### PR DESCRIPTION
**Pull request by gpleiss**
This addresses an issue pointed out in #218 

cc/ @Balandat @jacobrgardner

### CODE REVIEW

File: test/examples/test_kissgp_additive_regression.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 from torch.autograd import Variable
 from gpytorch.kernels import RBFKernel, AdditiveGridInterpolationKernel
 from gpytorch.likelihoods import GaussianLikelihood
-from gpytorch.means import ConstantMean
+from gpytorch.means import ZeroMean
```
Comment (Balandat) [inline test/examples/test_kissgp_additive_regression.py]: should we run the test against both ZeroMean and ConstantMean?

File: test/examples/test_kissgp_additive_regression.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 from torch.autograd import Variable
```
Comment (Balandat) [inline test/examples/test_kissgp_additive_regression.py]: Variable is deprecated - would you mind mind removing it from this example while you're working on this?

File: test/examples/test_kissgp_additive_regression.py (hunk 1 above)
Comment (gpleiss) [reply test/examples/test_kissgp_additive_regression.py]: All of our other tests use ConstantMean - I just wanted there to be one test with ZeroMean.

**Comment by gpleiss** [comment]
@Balandat added unit tests, removed variable

File: test/examples/test_kissgp_additive_regression.py [hunk 3]
```diff
@@ -75,12 +77,16 @@ def test_kissgp_gp_mean_abs_error(self):
         gp_model = GPRegressionModel(train_x.data, train_y.data, likelihood)
         mll = gpytorch.mlls.ExactMarginalLogLikelihood(likelihood, gp_model)
 
-        with gpytorch.settings.max_preconditioner_size(10), gpytorch.settings.max_cg_iterations(30):
+        with gpytorch.settings.max_preconditioner_size(
```
Comment (Balandat) [inline test/examples/test_kissgp_additive_regression.py]: this is pretty ugly - is this what black suggests?

**Comment by jacobrgardner** [comment]
I've been running `black -l 120` for what it's worth, I much prefer 120 chars per line to 80

File: test/means/test_multitask_mean.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        a = torch.Tensor([[[1, 2], [1, 2], [2, 4]], [[2, 3], [2, 3], [1, 3]]])
+        res = self.mean(a)
+        self.assertEqual(tuple(res.size()), (2, 3, 4))
+        self.assertTrue(res[:, :, 0].eq(5).all())
+        self.assertTrue(res[:, :, 1].eq(0).all())
+        self.assertTrue(res[:, :, 2].eq(0).all())
+        self.assertTrue(res[:, :, 3].eq(7).all())
+
+
+class TestMultitaskMeanSameMean(unittest.TestCase):
```
Comment (Balandat) [inline test/means/test_multitask_mean.py]: Can we add another test function in here that makes sure we're actually deepcopying the mean? That is, make sure "same mean" here means that this is a notational convenience rather than actually having the same mean for all tasks. Basically, just fill the data values for the means of a couple of different tasks with different values, and make sure the outputs are correct and distinct.

**Comment by Balandat** [comment]
yeah `black -l 120` makes sense, this with clause just comes out looking pretty weird...

**Comment by Balandat** [comment]
test failure seems unrelated, but can you still lock the seed to a different value so this passes?

**Comment by gpleiss** [comment]
@Balandat changed the seed - tests pass.
